_delphi_utils_python/data_proc/geomap/geo_data_proc.py

```diff
@@ -548,6 +543,99 @@ def derive_chngfips_state_crosswalk():
     group_to_state.to_csv(join(OUTPUT_DIR, CHNGFIPS_STATE_OUT_FILENAME), index=False)
 
 
+def fetch_county_groups_spreadsheet():
+    # County mapping file is derived from
+    # https://docs.google.com/spreadsheets/d/1PEce4CjjHbRM1Z5xEMNI6Xsq_b2kkCh0/edit#gid=871427657
+    sheet_id = "1PEce4CjjHbRM1Z5xEMNI6Xsq_b2kkCh0"
+    sheet_name = "groupings"
+    # Request sheet in CSV format via tag in URL.
+    url = f"https://docs.google.com/spreadsheets/d/{sheet_id}/gviz/tq?tqx=out:csv&sheet={sheet_name}"
+
+    county_groups = pd.read_csv(
+            url, dtype="string", index_col=False
+        ).dropna(
+            how="all", axis=1
+        )
+    county_groups["state FIPS"] = county_groups["state FIPS"].astype(int)
+
+    # Counties belonging to each group are listed (as FIPS codes) in the "county
+    # FIPS grouping" column, concatenated and separated by the pipe "|". Each
+    # included FIPS code is also listed in its own unnamed column. County groups
+    # can contain varying numbers of counties, and if CHNG provides new or updated
+    # county groupings the number of unnamed columns listing included counties
+    # could change. Use a general approach to find all of them.
+    county_cols = [colname for colname in county_groups.columns if colname.startswith("Unnamed: ")]
+
+    new_names = {
+        "state FIPS": "state_fips",
+        "county FIPS grouping": "fips_list",
+        **{ colname: ("county" + str(i + 1)) for i, colname in enumerate(county_cols)}
+    }
+
+    county_groups = county_groups.rename(
+            columns=new_names
+        )[new_names.values()]
+
+    return county_groups
+
+
+def assign_county_groups():
+    county_groups = fetch_county_groups_spreadsheet()
+
+    # If a `lowpop_county_groups.csv` already exists in `data_proc/geomap`, we
+    # have to be careful to not reassign a group number to a different group.
+    # Group numbers must remain fixed, even if a given county group is no longer
+    # being used.
```

Comment:
suggestion: add tests for this or defer to when it's needed

I'm glad you thought about how to handle updating the group mapping, but the procedure is sufficiently subtle that I'd feel better about it if we had tests for it, and that would mean figuring out how to write and run tests for code that's not in the actual python module. That sounds like work that won't be needed for a while though, so we could also decide to leave handling the already-exists case aside until we actually have a group-change situation to handle: either drop the code from this PR or throw an exception so we don't run it by mistake.

Reply:
vote: keep the old+new groups merging code so we have a starting point if there's a change in the future, but by default throw an exception if `not county_groups.empty`. 

(agree with Katie: this code is hard to reason about without having a REPL and the data open, but unit testing this is probably not worth it, since a small change in their format could break our code)